Tests/SwiftDocCTests/Infrastructure/PathHierarchyTests.swift

```diff
@@ -4051,7 +4051,54 @@ class PathHierarchyTests: XCTestCase {
             XCTFail("Unexpected error \(error)")
         }
     }
-    
+
+    func testMissingParentInSomeSymbolGraphs() throws {
+        let parentID = "some-parent"
+        let childID = "some-child"
+        let relationships: [SymbolGraph.Relationship] = [
+            .init(source: childID, target: parentID, kind: .memberOf, targetFallback: nil)
+        ]
+
+        let catalog = Folder(name: "unit-test.docc", content: [
+            JSONFile(name: "ModuleName-Platform1.symbols.json", content: makeSymbolGraph(
+                moduleName: "ModuleName",
+                symbols: [
+                    makeSymbol(
+                        id: childID,
+                        kind: .func,
+                        pathComponents: ["SomeParent", "SomeChild"],
+                        availability: [makeAvailabilityItem(domainName: "Platform1")]
+                    )
+                ],
+                relationships: relationships
```

Comment:
I don't think this is a valid input. This symbol graph has a relationship to a symbol that doesn't exist in that same symbol graph file.

Reply:
Right, i found one of these invalid symbol graphs in the wild and i wrote this PR to guard against crashing since the _group_ of symbol graphs is valid when all put together.

Reply:
In that case it would be good if this test explained that on its own symbol graph file is invalid, and shouldn't work, but in aggregate the two symbol graphs files are valid. 

In this specific test, we get the same outcome (as far as the path hierarchy is concerned) using only the 2nd valid symbol graph file. The invalid symbol graph doesn't contribute anything of value.